src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -136,6 +136,19 @@ class IRSuite extends SparkSuite {
     assertEvalsTo(If(True(), NA(TInt32()), I32(7)), null)
   }
 
+  @Test def testIfWithDifferentRequiredness() {
```

Comment:
Can we also test arrays?

Reply:
I think this is testing arrays.

I also tested the python code that triggered the crash in the issue, which worked.

I can add that as a python test, but this test does isolate the behavior

Reply:
Oops, you're totally right.  Can we also test structs?  :-)

Reply:
ah, yeah.